IHP/Server.hs

```diff
@@ -52,15 +53,17 @@ run configBuilder = do
 
     sessionMiddleware <- initSessionMiddleware sessionVault frameworkConfig
     staticMiddleware <- initStaticMiddleware frameworkConfig
+    let corsMiddleware = initCorsMiddleware frameworkConfig
     let requestLoggerMiddleware = get #requestLoggerMiddleware frameworkConfig
 
     let run = withBackgroundWorkers frameworkConfig $
-            runServer frameworkConfig $
+           runServer frameworkConfig $
                 staticMiddleware $
-                    sessionMiddleware $
-                            requestLoggerMiddleware $
-                                    methodOverridePost $
-                                        application
+                    corsMiddleware $
+                        sessionMiddleware $
+                                requestLoggerMiddleware $
+                                        methodOverridePost $
+                                            application
```

Comment:
maybe we should refactor this into piping now? It's getting continuously longer and harder to read, I think.

```haskell
let run = application
        |> methodOverridePost
        |> requestLoggerMiddleware
        |> sessionMiddleware
        |> ...
```

Reply:
But then this would display the request-response flow in a wrong order

Reply:
Ah, that's an angle I hadn't thought about yet. I always imagine middleware as a wrapper, and see it as layers added. But this way of thinking makes the current way of writing it clear. Refactored my thought-process, not the code: great! :D